Components/Coroutines/Town/TrinityTownRun.TransmuteMaterials.cs

```diff
@@ -114,48 +115,49 @@ public static async Task<bool> TransmuteMaterials(List<ItemSelectionType> types)
             var allowedTypes = settingsTypes.Select(GetCurrencyType);
             var otherType = orderedTypes.Skip(1).FirstOrDefault(t => allowedTypes.Contains(t));
 
-            if (!await TransmuteMaterials(highestCount, otherType))
-                return false;
+            if (await TransmuteMaterials(highestCount, otherType) == CoroutineResult.Running)
+                return CoroutineResult.Running;
 
             s_logger.Verbose($"[{nameof(TransmuteMaterials)}] CubeItemsToMaterials Finished!");
-            return true;
+            return CoroutineResult.Done;
         }
 
         /// <summary>
         /// Converts crafting materials into other types of crafting materials
         /// </summary>
         /// <param name="to">the type of material you will get more of</param>
-        public static async Task<bool> TransmuteMaterials(CurrencyType to)
+        public static async Task<CoroutineResult> TransmuteMaterials(CurrencyType to)
         {
             foreach (var currency in GetOtherCurrency(to).OrderByDescending(c => Core.Inventory.Currency.GetCurrency(c)))
             {
-                await TransmuteMaterials(currency, to);
+                if (await TransmuteMaterials(currency, to) == CoroutineResult.Running)
+                    return CoroutineResult.Running;
```

Comment:
This looks weird to me. We are breaking the foreach loop for Running condition. It would make more sense to have `while (await TransmuteMaterials(currency, to) == CoroutineResult.Running)`  loop here and `await Coroutine.Yield()` in the body to run it again.

I think most of the Running conditions should be handled that way with a timeout or max execution fail-safe.